### PR TITLE
migrate to Go Module, use official compiler for cross-compiling

### PR DESCRIPTION
**Pull request by QuantumGhost**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,16 +1,22 @@
+PLATFORMS := darwin/386 darwin/amd64 linux/386 linux/amd64 linux/arm \
```
Comment (CatTail) [inline Makefile]: I don't think the current code base support windows platform, there is a separate PR open to add support for windows platform https://github.com/git-hooks/git-hooks/pull/23

File: Makefile [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
+	go build -o build/git-hooks
 
-get:
-	go get github.com/tools/godep
-	godep restore ./...
+.PHONY: test build release $(PLATFORMS) clean
 
-.PHONY: test clean build get
+clean:
+	find build -name 'git-hooks_*' -delete
```
Comment (CatTail) [inline Makefile]: any reason not use `rm -rf build/*`?

File: Readme.md [hunk 3]
```diff
@@ -5,6 +5,10 @@
 
 Rewritten from [icefox/git-hooks](https://github.com/icefox/git-hooks), with extra features
 
+## Supported Go versions
+
+git-hooks supports the latest two Go version. (Currently 1.12 and 1.13)
```
Comment (CatTail) [inline Readme.md]: This is actually implementation details right? final user, which will use the binary directly do not care about which go version we use for development.

File: cli.go [hunk 4]
```diff
@@ -205,7 +208,7 @@ func uninstallGlobal() {
 
 // Check latest version of git-hooks by github release
 // If there are new version of git-hooks, download and replace the current one
-func update(releases []github.RepositoryRelease) {
+func update(releases []*github.RepositoryRelease) {
```
Comment (CatTail) [inline cli.go]: thanks for optimize this, I'm definitely just a Golang amateur and this is my first Golang project 😄

File: Readme.md (hunk 3 above)
Comment (QuantumGhost) [reply Readme.md]: Users may want to install git-hooks via `go get`, so I guess it's better to keep it.

File: Makefile (hunk 2 above)
Comment (QuantumGhost) [reply Makefile]: personal preference ;)
changed to `rm -rf build/*`

**Pull request merged**